internal/eva/settings.go

```diff
@@ -0,0 +1,123 @@
+package eva
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+const (
+	getSettingPath   = "/api/core/management/GetSetting"
+	setSettingPath   = "/api/core/management/SetSetting"
+	unsetSettingPath = "/api/core/management/UnsetSetting"
+)
+
+type SetSettingsRequest struct {
+	Key                string `json:"Key"`
+	Value              string `json:"Value"`
+	OrganizationUnitID int64  `json:"OrganizationUnitID,omitempty"`
+}
+
+type SetSettingsResponse struct {
+}
+
+func (c *Client) SetSettings(ctx context.Context, req SetSettingsRequest) (*SetSettingsResponse, error) {
+	resp, err := c.restClient.R().
+		SetBody(req).
+		Post(setSettingPath)
+
+	if err != nil {
+		tflog.Error(ctx, "An network error ocurred.", err)
+
+		return nil, err
+	}
+
+	if resp.StatusCode() != 200 {
+		tflog.Info(ctx, "Request failed", "Status code", resp.StatusCode(), "body", resp.String())
+
+		return nil, errors.New(fmt.Sprintf("Request failed with error: %s", resp.String()))
```

Comment:
`resp.String()` converts the whole response to string or only the `body` ? 

I found this https://stackoverflow.com/questions/38673673/access-http-response-as-string-in-go, not sure if it gives same result